test/e2e/storage/testsuites/volume_attachment.go

```diff
@@ -0,0 +1,232 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testsuites
+
+import (
+	"crypto/sha256"
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	csiv1alpha1 "k8s.io/csi-api/pkg/apis/csi/v1alpha1"
+	csiclient "k8s.io/csi-api/pkg/client/clientset/versioned"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/storage/drivers"
+	"k8s.io/kubernetes/test/e2e/storage/testpatterns"
+)
+
+type volumeAttachmentTestSuite struct {
+	tsInfo TestSuiteInfo
+}
+
+var _ TestSuite = &volumeAttachmentTestSuite{}
```

Comment:
I think we should not add these tests to the general suite.  These tests are more like integration tests, it's checking for implementation details that users do not need to care about.  @jsafrane wdyt?

Reply:
I want to make sure that A/D controller and kubelet work together to bring up a volume with various CSIDrivers. IMO, this is not possible in integration tests.

Reply:
@jsafrane 

I create a separate PR for volumeAttachment test. PTAL
https://github.com/kubernetes/kubernetes/pull/69469

It is only enabled for in-tree driver and will be just skipped until volumeAttachment is implemented for in-tree driver.

Reply:
Do we need to explicitly test the creation of VolumeAttachment objects for every CSI Plugin though?  Would it be sufficient to just test that you can launch a pod and read/write to it against any CSI driver?  The CSI driver deployment for non-attachable plugins can leave out the external-attacher sidecar to test this.

Reply:
> Do we need to explicitly test the creation of VolumeAttachment objects for every CSI Plugin though?

No, it's enough for hostpath CSI driver. The tests creates various CSIDrivers for it. It does not make sense to test it with GCE driver.